### PR TITLE
Fix clean blocks with dead context on debugger

### LINKED ISSUE #13779: [Clean Blocks] Test failing with clean blocks enabled: Debugger #13755
When turning on the clean block option in the compiler and recompile all, these tests related to the Debugger fail:

- testRecompileMethodToInBlockContextWithDeadHomeRemainsInOldBlock
- testRecompileMethodToInBlockContext
- testRecompileMethodToInBlockContextWithDeadHome

### PR DESCRIPTION
**Pull request by PalumboN**
Fix https://github.com/pharo-project/pharo/issues/13779

- A dead context is created for CleanBlocks in case it is not found in the stack
- Then small bugs appear -> check comment in `sourceNodeForPC:`
- I also doubt how `BytecodeEncoder >> isCreateBlockAt:in:` was resolved.

### Related PR: https://github.com/pharo-spec/NewTools/pull/605

### CODE REVIEW

**Comment by Ducasse** [comment]
Some unrelated tests are failing

**Comment by MarcusDenker** [comment]
the failing tests look somehow related, as they are about blocks (but in Sindarin)

```
unix-64 / Tests-unix-64 / testMoveToNodeWhenNodeIsInBlockThatCreatesContextAndBlockHasBeenCreated – Unix64.Sindarin.Tests.Base.SindarinDebuggerTest
unix-64 / Tests-unix-64 / testMoveToNodeWhenNodeIsInBlockThatCreatesContextAndBlockHasBeenCreatedBackward – Unix64.Sindarin.Tests.Base.SindarinDebuggerTest
unix-64 / Tests-unix-64 / testMoveToNodeWhenNodeIsNonInlinedAndEmbeddedInNonInlinedBlock – Unix64.Sindarin.Tests.Base.SindarinDebuggerTest
```

**Comment by jecisc** [comment]
tests passed!

**Comment by PalumboN** [comment]
Tests (Morphic) on OSX are crashing but I cannot reproduce it 👎

**Comment by jecisc** [comment]
This is a known issue that I hope will be workaround with: https://github.com/pharo-spec/Spec/pull/1467

all tests passed here

File: src/OpalCompiler-Core/CompiledBlock.extension.st [hunk 1]
```diff
@@ -12,9 +12,12 @@ CompiledBlock >> sourceNode [
 
 { #category : '*OpalCompiler-Core' }
 CompiledBlock >> sourceNodeForPC: aPC [
+
 	| blockNode |
 	blockNode := self outerCode sourceNodeForPC: self pcInOuter.
-	^blockNode sourceNodeForPC: aPC
+	"Bug in the cache? The mapping is returning Return node instead of Block"
```
Comment (MarcusDenker) [inline src/OpalCompiler-Core/CompiledBlock.extension.st:18]: Looks like it... I will check and add an issue for this

**Review by MarcusDenker — COMMENTED** [review]
Nice!

File: src/Kernel-BytecodeEncoders/EncoderForSistaV1.class.st [hunk 2]
```diff
@@ -469,7 +469,9 @@ EncoderForSistaV1 class >> isCreateBlockAt: pc in: method [
 
 	| byte |
 	byte := self nonExtensionBytecodeAt: pc in: method.
-	^ byte = 250
+	^ byte = 250 or: [
+		  (method sourceNodeForPC: pc) in: [ :node | "Clean blocks are created as PushConstant"
```
Comment (MarcusDenker) [inline src/Kernel-BytecodeEncoders/EncoderForSistaV1.class.st:473]: Falling back to the AST here is not that nice as it breaks the abstraction (the idea is that the bytecode level abstractions should be usable without the AST). But we could merge it for now and improve later, this is used only the the debugger which needs the AST anyway.

**Review by MarcusDenker — APPROVED** [review]

**Pull request merged**